packages/flutter/lib/src/foundation/assertions.dart

```diff
@@ -3,19 +3,114 @@
 // found in the LICENSE file.
 
 import 'basic_types.dart';
+import 'diagnostics.dart';
 import 'print.dart';
 
 /// Signature for [FlutterError.onError] handler.
 typedef FlutterExceptionHandler = void Function(FlutterErrorDetails details);
 
 /// Signature for [FlutterErrorDetails.informationCollector] callback
 /// and other callbacks that collect information into a string buffer.
-typedef InformationCollector = void Function(StringBuffer information);
+typedef InformationCollector = void Function(List<DiagnosticsNode> information);
```

Comment:
This isn't critical but in the spirit of removing technical debt I would suggest this refactor as there appear to be basically zero uses of this outside of package:flutter so there is no reason not to modernize it to be consistent and simple.

Reply:
probably fine (make sure to mention this in the breaking change e-mail as well).

For background, the information collector is mostly about building up data on demand, so that we don't have to actually build up all the data in cases where the information is never rendered. It's also used in places where we need information from parts of the codebase that aren't the parts actually calling 'throw'.

Reply:
Yes I'll call this out in the breaking change e-mail as well. The fix will be easy and it will keep this change from bloating the APIs just to avoid breaking changes.